packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+          return true;
+        });
+
+        utils.sendMessage('fullScreenChange');
+        expect(handlerInvoked).toBe(true);
+      });
+
+      it(`pages.registerFullScreenHandler should not invoke full screen handler when set to false initialized with ${context} context`, async () => {
+        await utils.initializeWithContext(context);
+
+        let handlerInvoked = true;
+        pages.registerFullScreenHandler((x: boolean) => {
+          handlerInvoked = false;
+          return false;
+        });
+
+        utils.sendMessage('fullScreenChange');
+        expect(handlerInvoked).toBe(false);
       });
-      const messageForRegister = utils.findMessageByFunc('registerHandler');
-      expect(messageForRegister).not.toBeNull();
-      expect(messageForRegister.args.length).toBe(1);
-      expect(messageForRegister.args[0]).toBe('focusEnter');
     });
+  });
+
+  describe('Testing pages.isSupported function', () => {
+    it('pages.isSupported should return false if the runtime says pages is not supported', () => {
+      utils.setRuntimeConfig({ apiVersion: 1, supports: {} });
+      expect(pages.isSupported()).not.toBeTruthy();
+    });
+
+    it('pages.isSupported should return true if the runtime says pages is supported', () => {
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
+      expect(pages.isSupported()).toBeTruthy();
+    });
+  });
+
+  describe('Testing pages.tabs namespace', () => {
+    describe('Testing pages.tabs.navigateToTab function', () => {
+      const tabInstance: TabInstance = {
+        tabName: 'MockTab',
+        internalTabInstanceId: 'MockTabInstanceId',
+        lastViewUnixEpochTime: null,
+        entityId: 'MockEntityId',
+        channelId: 'MockChannelId',
+        channelName: 'MockChannelName',
+        channelIsFavorite: true,
+        teamId: 'MockTeamId',
+        teamName: 'MockTeamName',
+        teamIsFavorite: true,
+        groupId: 'MockGroupID',
+        url: 'http://some-valid-content-url.com',
+        websiteUrl: 'http://some-valid-website-url.com',
+      };
+      it('pages.tabs.navigateToTab should not allow calls before initialization', async () => {
+        await expect(() => pages.tabs.navigateToTab(null)).rejects.toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`pages.tabs.navigateToTab should register the navigateToTab action when initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          pages.tabs.navigateToTab(tabInstance);
+          const navigateToTabMsg = utils.findMessageByFunc('navigateToTab');
+          expect(navigateToTabMsg).not.toBeNull();
+          expect(navigateToTabMsg.args[0]).toBe(tabInstance);
+        });
+        it(`pages.tabs.navigateToTab should register the navigateToTab action when initialized with ${context} context - success case`, async () => {
```

Comment:
I don't see a test case that validates that an error is thrown with message: "Invalid internalTabInstanceId and/or channelId were/was provided"

Reply:
Added in the latest commit